keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepository.kt

```diff
@@ -150,6 +160,26 @@ interface ResourceRepository : PeriodicallyCheckedRepository<ResourceHeader> {
     return true
   }
 
+  fun Resource<*>.toResourceSummary() =
+    ResourceSummary(
+      id = id,
+      kind = kind,
+      status = getStatus(id), // todo eb: this will become expensive
```

Comment:
This makes another database call to calculated the status. When this becomes a problem (or soon) I think I'll just create a table for the status and every time we write a history event for a resource we then update the status.

Reply:
We can't just read a single status, mainly to detect flapping (10 events which only contain 'delta detected' then 'actuation launched', in pairs). Is that the best way to determine this? Probably not. But that's what it's doing now.

Reply:
We could probably optimize by first fetching one event, then if needed fetching 10. I'll discuss this with you offline and make any changes in a separate PR.